x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -471,6 +471,29 @@ public static long chronoToLong(long dateTime, ChronoField chronoField, ZoneId z
         return Instant.ofEpochMilli(dateTime).atZone(zone).getLong(chronoField);
     }
 
+    /**
+     * Extract the given {@link ChronoField} value from a date specified as a long number of nanoseconds since epoch
+     * @param dateNanos - long nanoseconds since epoch
+     * @param chronoField - The field to extract
+     * @param zone - Timezone for the given date
+     * @return - long representing the given ChronoField value
+     */
+    public static long chronoToLongNanos(long dateNanos, BytesRef chronoField, ZoneId zone) {
```

Comment:
Optional: this method seems very specific, the `BytesRef` conversion could also be done in the processor. But even if kept here, once having a `ChronoField`, it would be good to reuse the `chronoToLongNanos(long, ChronoField, ZoneId)` below.